dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/User.java

```diff
@@ -773,33 +774,21 @@ public Set<OrganisationUnit> getDataViewOrganisationUnitsWithFallback() {
   // Logic - tei search organisation unit
   // -------------------------------------------------------------------------
 
-  public boolean hasTeiSearchOrganisationUnit() {
+  private boolean hasTeiSearchOrganisationUnit() {
     return !CollectionUtils.isEmpty(teiSearchOrganisationUnits);
   }
 
-  public OrganisationUnit getTeiSearchOrganisationUnit() {
-    return CollectionUtils.isEmpty(teiSearchOrganisationUnits)
-        ? null
-        : teiSearchOrganisationUnits.iterator().next();
-  }
-
-  public boolean hasTeiSearchOrganisationUnitWithFallback() {
-    return hasTeiSearchOrganisationUnit() || hasOrganisationUnit();
-  }
-
-  /**
-   * Returns the first of the tei search organisation units associated with the user. If none,
-   * returns the first of the data capture organisation units. If none, return nulls.
-   */
-  public OrganisationUnit getTeiSearchOrganisationUnitWithFallback() {
-    return hasTeiSearchOrganisationUnit() ? getTeiSearchOrganisationUnit() : getOrganisationUnit();
-  }
-
   /** Returns the tei search organisation units or organisation units if not exist. */
   public Set<OrganisationUnit> getTeiSearchOrganisationUnitsWithFallback() {
     return hasTeiSearchOrganisationUnit() ? teiSearchOrganisationUnits : organisationUnits;
   }
 
+  /** Returns all search organisation units and organisation units. */
```

Comment:
I think this javadoc would benefit from the explanation you give in the PR or issue. Maybe

`Users capture scope org units can be outside of the search scope org units. The effective search org units represent their union.
`
Then maybe something about when this should be used over the other methods. The other related org unit methods could also link to this one in their javadoc.